lib/_internal/_placehold/index.js

```diff
@@ -16,8 +16,7 @@ export default function placehold(fn, length, oldArgs) {
         length,
         oldArgs
           .reduce(function(_next, a) {
-            _next.push(a !== __ ? a : newArgs.length ? newArgs.shift() : __)
-            return _next
+            return _next.concat(a !== __ ? a : (newArgs.shift() || __))
```

Comment:
![image](https://user-images.githubusercontent.com/2991847/36509993-9ab0cf58-1772-11e8-8d21-c33d1ff31859.png)
Is it acceptable?